proxy/src/bind.rs

```diff
@@ -363,26 +389,40 @@ impl<B: tower_h2::Body + 'static> tower::Service for Binding<B> {
 
 
 impl Protocol {
+    /// Returns true if the request was originally received in absolute form.
+    pub fn was_absolute_form(&self) -> bool {
```

Comment:
i'd move to this down lower near `can_reuse_clients`... 

Generally prefer to start `impl` with the constructor. Also, this function is very similar to can_reuse_clients